app/controllers/vendor_api/vendor_api_controller.rb

```diff
@@ -15,10 +18,30 @@ def parameter_missing(e)
       render json: { errors: [{ error: 'ParameterMissing', message: e }] }, status: 422
     end
 
-  private
-
     def set_cors_headers
       headers['Access-Control-Allow-Origin'] = '*'
     end
+
+    def confirm_valid_api_token!
```

Comment:
Not sure about `confirm` here — we check for presence at validity. Would `require_api_token!` or `require_valid_api_token!` or sth be more expressive?

Reply:
I am kind of happy with `confirm_valid_api_token!`  since it confirms that the api token is valid,
other possible names:
`authorize_by_api_token!`,
'validate_api_token!`

Reply:
Replaced it by `require_valid_api_token!` in the end.